src/sipTransportConstructor.ts

```diff
@@ -76,8 +81,11 @@ var computeRandomTimeout = (
     return randomInterval + retryOffset;
 };
 
-async function __connect(this: WebPhoneSIPTransport, options?: any): Promise<void> {
-    return await this.__connect(options).catch(async err => {
+async function __connect(this: WebPhoneSIPTransport, options?: any): Promise<void>{
+    return await this.__connect(options).catch(async (err)=>{
+        this.emit('wsConnectionError');
```

Comment:
This exception is very difficult to track/capture and i couldnt generate this exception during my testing. Its necessary to listen to this event for the client if occurs. SDK is anyways handling this by disposing the ws listners and reconnecting , but client needs a way to listen to this.

Reply:
I meant just do

```js
this.emit('wsConnectionError', err);
```

Reply:
Fixed